chained/water_version_control.go

```diff
@@ -32,24 +32,48 @@ func newWaterVersionControl(dir string) *waterVersionControl {
 // Remember to Close the io.ReadCloser after using it
 func (vc *waterVersionControl) GetWASM(ctx context.Context, transport string, downloader waterWASMDownloader) (io.ReadCloser, error) {
 	path := filepath.Join(vc.dir, transport+".wasm")
-	var f io.ReadCloser
 	log.Debugf("trying to load file %q", path)
 	f, err := os.Open(path)
 	if err != nil && !os.IsNotExist(err) {
 		return nil, log.Errorf("failed to open file %s: %w", path, err)
 	}
 
 	if errors.Is(err, fs.ErrNotExist) || f == nil {
-		f, err = vc.downloadWASM(ctx, transport, downloader)
+		if f != nil {
+			f.Close()
+		}
+		response, err := vc.downloadWASM(ctx, transport, downloader)
 		if err != nil {
 			return nil, log.Errorf("failed to download WASM file: %w", err)
 		}
+
+		return response, nil
+	}
+
+	fi, err := f.Stat()
+	if err != nil {
+		return nil, log.Errorf("failed loading file info")
+	}
+
+	if fi.Size() == 0 {
```

Comment:
How could this happen?  How do we know it's not going to happen again in the second `downloadWASM` call?  This seems like an unnecessary and confusing edge case to handle.  Am I missing something?

Reply:
This was happening without the locker/concurrency issues, I should update this so we can handle if the WASM wasn't loaded